feat(invite): Guest invites (#6247)
hammerandchisel/discord-api-docs: docs/resources/Guild.md

| nick?                         | ?string                                         | this user's guild nickname                                                                                                                                                                                                           |
| avatar?                       | ?string                                         | the member's [guild avatar hash](#DOCS_REFERENCE/image-formatting)                                                                                                                                                                   |
| roles                         | array of snowflakes                             | array of [role](#DOCS_TOPICS_PERMISSIONS/role-object) object ids                                                                                                                                                                     |

> | joined_at                     | ?ISO8601 timestamp                              | when the user joined the guild                                                                                                                                                                                                       |

Reviewer: this is a breaking change that will cause issues for bots, shouldn't this be part of a new api version?

> info
> In `GUILD_` events, `pending` will always be included as true or false. In non `GUILD_` events which can only be triggered by non-`pending` users, `pending` will not be included.

> info

> > Member objects retrieved from `VOICE_STATE_UPDATE` events will have `joined_at: null` set on them if they were invited as a guest.

Reviewer 2: ```suggestion
> Member objects retrieved from `VOICE_STATE_UPDATE` events will have `joined_at` set as `null` if the member was invited as a guest.
```
